### PR TITLE
dev: add a page on ArduPilot's Unit Test suite

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: dev/source/docs/unit-tests.rst [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+Overview
+========
+
+The Unit test suite is run on ArduPilot's autotest server on most
+commits to the master branch, but can be run locally to vet software
+changes.  Adding tests is straight-forward and encouraged to prove a
+function's correctness after a change.
+
+
+Running AutoTest
```
Comment (khancyr) [inline dev/source/docs/unit-tests.rst:28]: unittest maybe ?

File: dev/source/docs/unit-tests.rst [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+.. note::
+
+   These tests work well under Linux.  cygwin64 is an unknown country.
+
+The unit tests are compiled using the `waf` build tool:
+
+::
+
+    pbarker@bluebottle:~/rc/ardupilot(master)$ ./waf configure --board=linux --debug
```
Comment (khancyr) [inline dev/source/docs/unit-tests.rst:44]: why linux ?

File: dev/source/docs/unit-tests.rst [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+    .
+    tests/test_polygon                                    204433   7056   1352  212841
+    tests/test_segment_intersection                       193839   5832   1288  200959
+    tests/test_vector2                                    210539   6472   1320  218331
+
+    Build commands will be stored in build/linux/compile_commands.json
+    'tests' finished successfully (2.249s)
+    pbarker@bluebottle:~/rc/ardupilot(master)$  
+
+You can then simply run the test from the build directory:
```
Comment (khancyr) [inline dev/source/docs/unit-tests.rst:68]: or use waf --alltests

**Review by khancyr — APPROVED** [review]
seems good !

File: dev/source/docs/unit-tests.rst (hunk 1 above)
Comment (peterbarker) [reply dev/source/docs/unit-tests.rst]: Fixed, thanks!

File: dev/source/docs/unit-tests.rst (hunk 2 above)
Comment (peterbarker) [reply dev/source/docs/unit-tests.rst:42]: They're native binaries - they don't involve simulated sensor drivers.

File: dev/source/docs/unit-tests.rst (hunk 3 above)
Comment (peterbarker) [reply dev/source/docs/unit-tests.rst:66]: Maybe?

I didn't know about that one... and I guess that could run things other than the unit tests?

File: dev/source/docs/unit-tests.rst [hunk 4]
```diff
@@ -0,0 +1,198 @@
+.. _ardupilot-unit-tests:
+
+====================
+ArduPilot Unit Tests
+====================
+
+ArduPilot's Unit Test suite allows for the creation of repeatable tests
+which help prevent regressions in ArduPilot's internal functions.  It is based
+on Google's `GTest Framework <https://https://github.com/google/googletest>`.
```
Comment (Hwurzburg) [inline dev/source/docs/unit-tests.rst:9]: link invalid...needs double underscore after last back tick

File: dev/source/docs/unit-tests.rst [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+==================
+
+The Unit Tests require an environment in which the ArduPilot executables can be compiled.  The easiest way to obtain this is to create a SITL testing environment.  Use the SITL instructions (:ref:`SITL <using-sitl-for-ardupilot-testing>`) to obtain a valid environment.  It is suggested the ArduPilot Vagrant virtual machine configuration files be used to obtain a working environement.
+
+Invocation
+----------
+
+.. note::
+
+   These tests work well under Linux.  cygwin64 is an unknown country.
```
Comment (Hwurzburg) [inline dev/source/docs/unit-tests.rst:36]: did not know that there was a country named cygwin64...or territory either for that matter....perhaps operation under Cygwin64 is untested?

**Review by Hwurzburg — CHANGES_REQUESTED** [review]

File: dev/source/docs/unit-tests.rst (hunk 4 above)
Comment (peterbarker) [reply dev/source/docs/unit-tests.rst]: Fixed.

File: dev/source/docs/unit-tests.rst (hunk 5 above)
Comment (peterbarker) [reply dev/source/docs/unit-tests.rst]: Fixed.

**Comment by Hwurzburg** [comment]
thanks....merging

**Pull request merged**